lib/spack/spack/build_systems/nmakefile.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import inspect
+from typing import List  # novm
+
+import llnl.util.filesystem as fs
+
+import spack.builder
+import spack.package_base
+from spack.directives import build_system, conflicts
+
+from ._checks import (
+    BaseBuilder,
+)
+
+
+class NMakePackage(spack.package_base.PackageBase):
+    """Specialized class for packages built using a Makefiles."""
+
+    #: This attribute is used in UI queries that need to know the build
+    #: system base class
+    build_system_class = "NMakePackage"
+
+    build_system("nmake")
+    conflicts("platform=linux",  when="build_system=nmake")
+    conflicts("platform=darwin", when="build_system=nmake")
+    conflicts("platform=cray",   when="build_system=nmake")
```

Comment:
There's already a `nmake.py` file. Is this different? In case can you extend the other one?